.github/workflows/build.yaml

```diff
@@ -0,0 +1,68 @@
+name: Build ADI br2-external
+
+on:
+  push:
+  pull_request:
+
+env:
+  BR2_DL_DIR: $HOME/.buildroot-dl
+  CC: gcc-15
+  CXX: g++-15
+
+jobs:
+  build:
+    runs-on: self-hosted
+
+    strategy:
+      fail-fast: false
+      matrix:
+        defconfig:
+          - adi_sc598_ezkit_defconfig
+
+    steps:
+      - name: Checkout br2-external
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Prepare Buildroot download dir
+        run: mkdir -p "$BR2_DL_DIR"
+
+      - name: Cache Buildroot downloads
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.BR2_DL_DIR }}
+          key: ${{ runner.os }}-buildroot-dl-${{ hashFiles('configs/**/*.fragment', 'configs/**/*.defconfig') }}
+          restore-keys: |
+            ${{ runner.os }}-buildroot-dl-
+
+      - name: Build image for ${{ matrix.defconfig }}
+        run: |
+          cd buildroot
+
+          echo ">>> Selecting defconfig: ${{ matrix.defconfig }}"
+
+          make BR2_EXTERNAL="${GITHUB_WORKSPACE}" ${{ matrix.defconfig }}
+
+          if [ -f "${GITHUB_WORKSPACE}/configs/buildroot.fragment" ] || \
+             [ -f "${GITHUB_WORKSPACE}/configs/debug.fragment" ]; then
+            support/kconfig/merge_config.sh .config \
+              "${GITHUB_WORKSPACE}/configs/buildroot.fragment" \
+              "${GITHUB_WORKSPACE}/configs/debug.fragment"
+          fi
+
+          make -j$(nproc)
+
+      - name: Upload Buildroot images for ${{ matrix.defconfig }}
+        if: success()
+        uses: actions/upload-artifact@v4
+        with:
+          name: images-${{ matrix.defconfig }}
+          path: buildroot/output/images
+
+      - name: Upload logs for ${{ matrix.defconfig }} (failure only)
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: logs-${{ matrix.defconfig }}
+          path: buildroot/output/build
```

Comment:
This isn't just logs, this is the whole build directory. And still missing critical things like the `.config`. On my local machine this is 12G. This should be removed unless there is a real need that outweighs the resource demand. I manually cancelled this in the latest run after it was running for 17m.